ionic2/app/components/map/map.component.ts

```diff
@@ -67,15 +68,22 @@ export class MapComponent {
       this.map.navigate(start, destination);
     })
   }
+}
 
-  private static mapPokePOI(pokePOI: Object): (PokeSighting | PokePrediction | PokeMob) {
-    if ('source' in pokePOI) {
-      return PokeSighting.fromObject(pokePOI);
-    } else if ('clusterId' in pokePOI) {
-      return PokeMob.fromObject(pokePOI);
-    } else {
-      throw new Error('PokePOI cannot be identified as PokeSighting or PokeMob:\n' + JSON.stringify(pokePOI));
-    }
+function poiFromMapEventData(rawData: any) : POI {
```

Comment:
TypeScript docu uses the keyword as well https://www.typescriptlang.org/docs/handbook/functions.html